auth/credentials/impersonate/user.go

```diff
@@ -153,6 +154,7 @@ func (u userTokenProvider) exchangeToken(ctx context.Context, signedJWT string)
 	if err != nil {
 		return nil, err
 	}
+	req.Header.Set(header.GOOGLE_API_CLIENT_HEADER, header.GetGoogHeaderToken(header.CredTypeImp, header.TokenTypeAccess))
```

Comment:
for normal user you had the type as none, but for an impersonated user it is access. Is that correct?

Reply:
@arithmetic1728 Should this [impersonate/user] flow be considered cred-type `u` or `imp`? I couldn't find a similar flow in googleapis/google-auth-library-python. Please link if you know of one.

Reply:
https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/impersonated_credentials.py#L425 is the python implementation for impersonation. I see `cred-type/imp` is used.

The token type needs to be `access` or `id` based on what token is being fetched/used. For normal user it could have been none because both the tokens are fetched at same time.